sources/Outputs/include/Dyd.h

```diff
@@ -74,20 +75,24 @@ class Dyd {
      * @param gens generators definition coming from algorithms
      * @param loaddefs load definitions coming from algorithms
      * @param slacknode the slack node to use
+     * @param hvdcLines hvdc definition coming from algorithms
      */
     DydDefinition(const std::string& base, const std::string& filepath, const std::vector<algo::GeneratorDefinition>& gens,
-                  const std::vector<algo::LoadDefinition>& loaddefs, const std::shared_ptr<inputs::Node>& slacknode) :
+                  const std::vector<algo::LoadDefinition>& loaddefs, const std::shared_ptr<inputs::Node>& slacknode,
+                  const std::vector<algo::HvdcLineDefinition>& hvdcLines) :
         basename{base},
         filename{filepath},
         generators{gens},
         loads{loaddefs},
-        slackNode{slacknode} {}
+        slackNode{slacknode},
+        hvdcLines{hvdcLines} {}
 
     std::string basename;                               ///< basename for file
     std::string filename;                               ///< filepath for file to write
     std::vector<algo::GeneratorDefinition> generators;  ///< generators found
     std::vector<algo::LoadDefinition> loads;            ///< list of loads
     std::shared_ptr<inputs::Node> slackNode;            ///< slack node to use
+    std::vector<algo::HvdcLineDefinition> hvdcLines;    ///< list of hvdc lines
```

Comment:
I just noticed that but do we really need to copy all those vectors? cannot we use a reference?

Reply:
I just tried to put all these vector as const vector& and the test passed, should I add it or drop it ? So we can do this for more attributes but in a separate PR that would fix a newly created issue ?

Reply:
It is working only because we call the function write of DydWriter in the same scope as the references variable. I think this is dangerous. If we want to avoid copy, the best way would be to use universal references and move

Reply:
Yes but where else would we call the method write of DydWriter ?

Reply:
we cannot move as the same vectors are used in Par right?

Reply:
correct, we cannot move. I don't like the idea of referencing local variables because we break the data encapsulation for a struct

Reply:
However, it could be a serious gain in time

Reply:
I would like a benchmark on this one. If there is a clear gain, OK, let's do it. If not, I would prefer we don't do this, since it breaks the original "spirit" of the structure

Reply:
If you benchmark it, change also the vector of the par

Reply:
Gain tested on Marco's IIDM is around 1ms (worst difference) to 3ms, with sometimes the version without the references better than the version with references. @rosiereflo whet do you think ?

Reply:
How much percentage gained is this 1ms ?

Reply:
this is another topic so maybe we can discuss about this in another ticket.